cmake/OpenCVCompilerOptimizations.cmake

```diff
@@ -347,22 +347,16 @@ elseif(ARM OR AARCH64)
     ocv_update(CPU_FP16_IMPLIES "NEON")
   else()
     ocv_update(CPU_KNOWN_OPTIMIZATIONS "NEON;FP16;NEON_DOTPROD;NEON_FP16;NEON_BF16")
-    ocv_update(CPU_NEON_FLAGS_ON "")
-    ocv_update(CPU_FP16_IMPLIES "NEON")
-    ocv_update(CPU_NEON_DOTPROD_IMPLIES "NEON")
-    ocv_update(CPU_NEON_FP16_IMPLIES "NEON")
-    ocv_update(CPU_NEON_BF16_IMPLIES "NEON")
-    if(MSVC)
-      ocv_update(CPU_NEON_DOTPROD_FLAGS_ON "")
-      ocv_update(CPU_NEON_FP16_FLAGS_ON "")
-      ocv_update(CPU_NEON_BF16_FLAGS_ON "")
-    else()
-      ocv_update(CPU_NEON_DOTPROD_FLAGS_ON "-march=armv8.2-a+dotprod")
-      ocv_update(CPU_NEON_FP16_FLAGS_ON "-march=armv8.2-a+fp16")
-      ocv_update(CPU_NEON_BF16_FLAGS_ON "-march=armv8.2-a+bf16")
+    set(CPU_BASELINE "DETECT" CACHE STRING "${HELP_CPU_BASELINE}")
+    set(CPU_DISPATCH "" CACHE STRING "${HELP_CPU_DISPATCH}")
+    if(NOT CPU_BASELINE STREQUAL "DETECT")
+      message(WARNING "ARM64 only supports CPU_BASELINE=DETECT, use CMAKE_CXX_FLAGS or cmake toolchain file options to change platform")
```

Comment:
`CPU_BASELINE=NATIVE` should be supported.

Reply:
Done.